### ISSUE TITLE
Not working with WSL2

### ISSUE THREAD

**Issue by patroza** [wallabyjs/public#2118]
### Issue description or question
Appears not to work with WSL2. The Wallaby output is empty in VSCode.
The node process appears to run though:

`node --harmony /mnt/c/Users/username/.vscode/extensions/wallabyjs.wallaby-vscode-1.0.128/wallaby/server.js extended-core-ws --projectCachePath=/mnt/c/Users/username/.vscode/extensions/wallabyjs.wallaby-vscode-1.0.128/projects/eb728f543f68f584 --configPath=/home/username/projects/fp-app-framework/wallaby.js --lkp=/mnt/c/Users/username/.wallaby/key.lic --client=VSCode --phantomjs=c:Usersusername.vscodephantomjs_v2.1.1phantomjs.exe --editorTypeScript=undefined --port=50685 --quokka= --homeDir=/mnt/c/Users/username --cv=undefined`

One suspect appears `--phantomjs=c:Usersusername.vscodephantomjs_v2.1.1phantomjs.exe`
there are also two `undefined` values in there.

I guess another concern is that at this time the port will be opened not on localhost but the WSL2 vm's ip address.

### Wallaby.js configuration file

In vs code settings.json:
`"wallaby.useWsl": true`

(config, works without WSL)
```javascript
module.exports = function (w) {
  return {
    files: [
      '**/*.ts',
      { pattern: '**/*.d.ts', ignore: true },
      { pattern: '**/*.test.ts', ignore: true },
      { pattern: 'node_modules', ignore: true }
    ],

    tests: [
      '**/*.test.ts',
      { pattern: 'node_modules', ignore: true }
    ],
    // for node.js tests you need to set env property as well
    // https://wallabyjs.com/docs/integration/node.html
    env: {
      type: 'node',
      runner: 'node'
    },

    testFramework: 'jest',
    compilers: {
      '**/*.ts?(x)': w.compilers.typeScript({ isolatedModules: true })
    }
  };
};
```

### Code editor or IDE name and version

Visual Studio Code v1.35.1

### OS name and version

Windows 10 Version 10.0.18917.1000

**Comment by patroza** [comment]
Tried to use `https://gist.github.com/sqren/2d9639ad228ccb051a9da752c8c70c66` to port forward `50685` to the WSL instance.

Then got output in Wallaby console:
```
[Error] Failed to set up project file system: Error: not opened 
[Error]     at _0xc1558e.emit (events.js:198:13) 
[Error]     at process._tickCallback (internal/process/next_tick.js:61:11) 
[Info]  wallaby.js stopped
```

**Comment by smcenlly** [comment]
We don't expect that WSL 2 will work without changes to wallaby because of the networking changes in WSL 2. As you said, WSL 2 now runs with its own IP address vs. using host IP in WSL 1. 

We will be adding support for WSL 2 but do not plan to do so until it's made available in the production branch or the Release Preview branch (at the earliest).

Until we add support for WSL 2, you may switch to use WSL 1 if you want to continue using Wallaby with WSL 1. From [Microsoft devblogs](https://devblogs.microsoft.com/commandline/wsl-2-is-now-available-in-windows-insiders/):

```
wsl --set-version <Distro> <Version>
```
>Use this command to convert a distro to use the WSL 2 architecture or use the WSL 1 architecture.
>
>: the specific Linux distro (e.g. “Ubuntu”)
>
>: 1 or 2 (for WSL 1 or 2)

RE: https://code.visualstudio.com/docs/remote/remote-overview, have you tried containers and ssh? We don't use these ourselves but did some testing when they were first released and everything seemed to be working for us. The only annoying thing was that because of container isolation, you had to add your wallaby license for each project. We tested using a combination of Windows and OSX so let us know if you can't get it working and we'll investigate for you.

**Comment by jim-pickering** [comment]
WSL2 is about to go GA this month, on version 2004; available now for free on Windows Insiders Slow Ring. I hope the required changes to Wallabyjs don't take too long. All of my development revolves around WSL2 and Docker. I'd rather not use Wallabyjs than go back to WSL, honestly.

**Comment by smcenlly** [comment]
@jim-pickering - we'll be looking into WSL2 over the next few weeks. We haven't yet started working on this yet but unless we encounter major technical problems, we expect to have full support for WSL2 in the next 1-2 months.

**Comment by smcenlly** [comment]
Today we released Wallaby v1.0.888, which now natively supports WSL2 using VS Code. 

There are a few simple steps you need to take to start using Wallaby with WSL2:
1. Open your WSL2 project in VS Code
2. In the VS Code Extensions View, install Wallaby for your WSL instance.
3. After activating (note: you'll probably be asked to reload your editor), reapply your Wallaby license using, "Wallaby.js: Manage License Key".

If your tests run using node.js (e.g. if you use Jest), everything should just work for you. 

If your tests run in the browser (i.e. chrome-headless) then you will need to ensure that your WSL instance can access your desktop IP Address by updating your Windows Firewall Rules. You may confirm everything works by temporarily disabling your Firewall before re-enabling and configuring. One option is to exclude the `vEthernet (WSL)` network from your list of `public` `protected network connections` (see screenshot below). We'll be updating our docs over the next day or so with more detailed instructions and will post a link to our docs when they're ready.

![image](https://user-images.githubusercontent.com/2074089/80450399-cd296b00-8964-11ea-8b3b-87d211b85a6d.png)

**Comment by smcenlly** [comment]
We have updated [our docs](https://wallabyjs.com/docs/intro/get-started-vscode.html#wsl-2) to describe working with WSL 2. We couldn't find a way to allow network access between the host and the WSL2 distribution other than what was described in our last update. It appears that there are network problems that the WSL team needs to address (https://github.com/microsoft/WSL/issues/4139, https://github.com/microsoft/WSL/issues/4619) for `WSL2 distribution -> Windows host` network communications to work out of the box.
